mitxgraders/baseclasses.py

```diff
@@ -169,6 +197,10 @@ def __call__(self, expect, student_input):
                     formatted = msg.format(student_input)
                 raise StudentFacingError(formatted)
 
+        # Handle partial credit based on attempt number
+        if self.config['attempt_based_credit']:
+            self.apply_attempt_based_credit(result, kwargs.get('attempt'))
```

Comment:
This time I have a justification for my single-use well-named function: somebody may want to shadow it!